app/src/main/java/org/thoughtcrime/securesms/util/VersionDataFetcher.kt

```diff
@@ -0,0 +1,59 @@
+package org.thoughtcrime.securesms.util
+
+import android.os.Handler
+import android.os.Looper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.session.libsession.messaging.file_server.FileServerApi
+import org.session.libsession.utilities.TextSecurePreferences
+import org.session.libsignal.utilities.Log
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.time.Duration.Companion.hours
+
+private val TAG: String = VersionDataFetcher::class.java.simpleName
+private val REFRESH_TIME_MS = 4.hours.inWholeMilliseconds
+
+@Singleton
+class VersionDataFetcher @Inject constructor(
+    private val prefs: TextSecurePreferences
+) {
+    private val handler = Handler(Looper.getMainLooper())
+    private val fetchVersionData = Runnable {
+        scope.launch {
+            try {
+                // Perform the version check
+                val clientVersion = FileServerApi.getClientVersion()
+                Log.i(TAG, "Fetched version data: $clientVersion")
+                prefs.setLastVersionCheck()
+                startTimedVersionCheck()
+            } catch (e: Exception) {
+                // We can silently ignore the error
+                Log.e(TAG, "Error fetching version data", e)
+                // Schedule the next check for 4 hours from now, but do not setLastVersionCheck
+                // so the app will retry when the app is next foregrounded.
+                startTimedVersionCheck(REFRESH_TIME_MS)
+            }
+        }
+    }
+
+    private val scope = CoroutineScope(Dispatchers.Default)
+
+    /**
+     * Schedules fetching version data [delayMillis] milliseconds from now.
+     *
```

Comment:
Could add this line to explain what delayMillis defaults to
`[delayMillis] defaults to 4h since the last check`

Reply:
@bemusementpark

Reply:
addressed.